pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -1222,9 +1228,22 @@ class Build {
             // Create the repo handler with the user's defaults to ensure a temurin-build checkout is not null
             def repoHandler = new RepoHandler(USER_REMOTE_CONFIGS)
             repoHandler.setUserDefaultsJson(context, DEFAULTS_JSON['defaultsUrl'])
+
+            /* 
+                if BUILD_REF, CI_REF are set in BUILD_CONFIGURATION, overwrite branche value in DEFAULTS_JSON
+                so we can re-use the same logic down: get config either from DEFAULT_JSON or ADPOT_DEFULAT_JSON(enable useAdoptShellScripts)
+            */
+            DEFAULTS_JSON['repository']['build_branch'] = buildConfig.BUILD_REF ?: DEFAULTS_JSON['repository']['build_branch']
```

Comment:
This doesn't seem right, logic further down checks for buildConfig.BUILD_REF being set otherwise uses DEFAULTS_JSON, so not sure why this is overriding DEFAULTS_JSON ..?
eg.line 1506: String helperRef = buildConfig.HELPER_REF ?: DEFAULTS_JSON['repository']['helper_ref']